File: app/src/main/java/com/tevinjeffrey/rutgersct/ui/courseinfo/CourseInfoSubcomponent.java
package com.tevinjeffrey.rutgersct.ui.courseinfo;

import com.tevinjeffrey.rutgersct.ui.sectioninfo.SectionInfoFragment;
import com.tevinjeffrey.rutgersct.ui.sectioninfo.SectionInfoPresenterImpl;
import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@Subcomponent
public interface CourseInfoSubcomponent
    extends AndroidInjector<CourseInfoFragment> {

  @Subcomponent.Builder
  abstract class Builder extends AndroidInjector.Builder<CourseInfoFragment> {
    @Override public abstract CourseInfoSubcomponent build();
  }
}

File: app/src/main/java/com/tevinjeffrey/rutgersct/ui/courseinfo/CourseInfoSubcomponent.kt
package com.tevinjeffrey.rutgersct.ui.courseinfo

import dagger.Subcomponent
import dagger.android.AndroidInjector

@Subcomponent
interface CourseInfoSubcomponent : AndroidInjector<CourseInfoFragment> {

  @Subcomponent.Builder
  abstract class Builder : AndroidInjector.Builder<CourseInfoFragment>() {
    abstract override fun build(): CourseInfoSubcomponent
  }
}
